packages/wonder-blocks-data/src/hooks/use-cached-effect.js

```diff
@@ -104,107 +120,152 @@ export const useCachedEffect = <TData: ValidCacheData>(
         // that all calls when the request is in-flight will update once that
         // request is done, we want the cache to be empty until that point.
     );
-
-    // Build a function that will update the cache and either invoke the
-    // callback provided in options, or force an update.
     const forceUpdate = useForceUpdate();
-    const setCacheAndNotify = React.useCallback(
-        (value: Result<TData>) => {
-            setMostRecentResult(value);
-
-            // If our caller provided a cacheUpdated callback, we use that.
-            // Otherwise, we toggle our little state update.
-            if (onResultChanged != null) {
-                onResultChanged(value);
-            } else {
-                forceUpdate();
+    // For the NetworkOnly fetch policy, we ignore the cached value.
+    // So we need somewhere else to store the network value.
+    const networkResultRef = React.useRef();
+
+    // Set up the function that will do the fetching.
+    const currentRequestRef = React.useRef();
+    const fetchRequest = React.useMemo(() => {
+        // We aren't using useCallback here because we need to make sure that
+        // if we are rememo-izing, we cancel any inflight request for the old
+        // callback.
+        currentRequestRef.current?.cancel();
+        currentRequestRef.current = null;
+        networkResultRef.current = null;
+
+        // Now we can return the new fetch function.
+        return () => {
+            if (fetchPolicy === FetchPolicy.CacheOnly) {
+                throw new DataError(
+                    "Cannot fetch with CacheOnly policy",
+                    DataErrors.NotAllowed,
+                );
             }
-        },
-        [setMostRecentResult, onResultChanged, forceUpdate],
-    );
+            // We use our request fulfillment here so that in-flight
+            // requests are shared.
+            const request = RequestFulfillment.Default.fulfill(requestId, {
+                handler: interceptedHandler,
+            });
+
+            if (request === currentRequestRef.current?.request) {
+                // The request inflight is the same, so do nothing.
+                // NOTE: Perhaps if invoked via a refetch, we will want to
+                // override this behavior and force a new request?
+                return;
+            }
+
+            // Clear the last network result.
+            networkResultRef.current = null;
+
+            // Cancel the previous request.
+            currentRequestRef.current?.cancel();
```

Comment:
So, a `refetch` occurs but the inflight request is different. I don't "think" this is technically possible since the callback itself would've been recreated, but being defensive to make sure.